### PR TITLE
Acorn update, fixes spread operator in object literal

### LINKED ISSUE #275: Spread operator in object literal causes SyntaxError
Any module with a spread operator in an object literal will cause a SyntaxError.
This is because [Acorn](https://github.com/acornjs/acorn) doesn't support the spread operator in object literals (https://github.com/acornjs/acorn/issues/388) and all modules are parsed by [Acorn](https://github.com/acornjs/acorn) since https://github.com/rollup/rollup-plugin-commonjs/commit/b9191ae30d1fce2e11da1f014a7a01aa119a77b6. 
Before that the `checkFirstpass` function provided an early exit. 

Reduced test case:

*index.js*
```js
const obj = { foo: 'bar' };
const clone = { ...obj };
```

*rollup.config.js*
```js
import commonjs from 'rollup-plugin-commonjs';

export default {
  input: 'index.js',
  output: {
    format: 'cjs',
  },
  plugins: [commonjs()],
};
```
Yields `[!] (commonjs plugin) SyntaxError: Unexpected token`

### PR DESCRIPTION
**Pull request by walterra**
Fixes #275.

- Updates acorn to 5.4.1
- Updates ecmaVersion to 9

This is similar to this PR for rollup https://github.com/rollup/rollup/pull/1953

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -20,7 +20,7 @@
   },
   "extends": "eslint:recommended",
   "parserOptions": {
-    "ecmaVersion": 8,
+    "ecmaVersion": 9,
```
Comment (adrianheine) [inline .eslintrc]: It's not necessary to bump the ECMAVersion eslint tries to parse, since it didn't change.

Comment (walterra) [reply .eslintrc]: Thanks, I reverted that file to use 8.

**Comment by walterra** [comment]
@adrianheine regarding the use of `parse`, I'm afraid I don't know that much about rollup+plugins to do that. Updating the existing dependencies is what helped me to get my fork working and I thought I'd just create a PR for that.

**Comment by KingHenne** [comment]
> @adrianheine regarding the use of parse, I'm afraid I don't know that much about rollup+plugins to do that. Updating the existing dependencies is what helped me to get my fork working with object spread operators and I thought I'd just create a PR for that.

@walterra #287 removes acorn as dependency and uses rollup's exposed `parse` function instead. I think this is a preferable solution to this PR.

**Comment by walterra** [comment]
Cool, closing this one.

**Pull request closed without merging**